### PR TITLE
Calculator v1.0 milestone5

### PR DESCRIPTION
**Pull request by Bismarck-GM**
### Milestone 5:

In this Milestone I:

- Implemented the Functions needed to pass props between components.
- Changed App function component to a Class Component and added states.
- Refactor Logic Code to work properly.
- Deployed the changes to Heroku.

Pushed the new build to Heroku. [You can check it by clicking here.](https://naked-calculator.herokuapp.com/)

Thanks for the review! 🌵

### CODE REVIEW

File: src/components/App.js [hunk 1]
```diff
@@ -1,15 +1,33 @@
...
+    const data = this.state;
+    this.setState(calculate(data, buttonName));
+  }
+
+  render() {
+    const data = this.state;
+    return (
+      <>
+        <Display result={data.next || data.total || '0'} />
+        <ButtonPanel clickHandler={this.handleClick} />
```
Comment (SpaYco) [inline src/components/App.js]: - [x] Use Destructuring 

```JavaScript
   const { next, total } = this.state; // Destructuring 
   return (
      <>
        <Display result={ next || total || '0'} /> // I called next and total without having to call data.
        <ButtonPanel clickHandler={this.handleClick} />
      </>
```

**Review by SpaYco — CHANGES_REQUESTED** [review]
# Changes Requested 

Hello! 👋 
Awesome job so far!! 👍 you still need to make some changes though! 

## Required Changes ♻️ 

- [ ] move `handleClick` inside the constructor

```JavaScript
constructor(props) {
  super(props);
  this.state = {
      total: null,
      next: null,
      operation: null,
  };
  }

  handleClick = (buttonName) => {
  this.setState(calculate(this.state, buttonName));
  }
}
```

### please make the appropriate changes and ask for another code review ✔️ 

Happy Coding!

Best, 
Aziz Mejri 
_contact me on [twitter](https://twitter.com/iSpaYco) or slack (@SpaYco) if you need any help_

File: src/components/App.js (hunk 1 above)
Comment (Bismarck-GM) [reply src/components/App.js]: This last one is fine, I just noticed it.

The first request isn't making much sense to me, how can I add the function inside the constructor?
The code given isn't the solution, and never do what you're asking, even though I'm trying to find how to implement the function inside of it, I can not find a clear answer.

Can you elaborate on that one?

**Review by christianotieno — APPROVED** [review]
### Milestone Approved :champagne:
Hello @Bismarck-GM 👋 
 ##
This milestone has met all the requirements needed for approval. 👏 
Congratulations on the hard work done on this milestone. Good job 👍 
I will go ahead and mark this as complete. Good luck with the next tasks 🥇 
##
### Happy coding 💻 
#### Christian Otieno (CR)

**Pull request merged**